docs/dev-guide/intellij-plugin.md

```diff
@@ -0,0 +1,187 @@
+| `textDocument/definition`                  | ✅ Implemented        | Handled by lsp4ij     | Go to definition functionality                                                                                                   |
+| `textDocument/signatureHelp`               | ✅ Implemented        | Handled by lsp4ij     | Function signature hints                                                                                                         |
+| `textDocument/rename`                      | ✅ Implemented        | Handled by lsp4ij     | Symbol renaming                                                                                                                  |
+| `textDocument/publishDiagnostics`          | ✅ Implemented        | Direct implementation | Custom diagnostic formatting with HTML support (TinymistLanguageClient.kt:24)                                                    |
+| `textDocument/semanticTokens`              | ✅ Implemented        | Handled by lsp4ij     | Semantic syntax highlighting                                                                                                     |
+| `textDocument/references`                  | ✅ Implemented        | Handled by lsp4ij     | Find all references to a symbol                                                                                                  |
+| `textDocument/documentHighlight`           | ✅ partly implemented | Handled by lsp4ij     | Highlight related symbols; currently the highlight only works upon entirely selecting a symbol not just placing the carret there |
+| `textDocument/documentSymbol`              | ✅ Implemented        | Handled by lsp4ij     | Document outline/structure view                                                                                                  |
+| `textDocument/inlayHint`                   | ✅ Implemented        | Handled by lsp4ij     | Inlay additional information into code editor, i.e. the names of function parameters                                             |
+| `textDocument/codeAction`                  | ✅ Implemented        | Handled by lsp4ij     | Code fixes and refactoring actions                                                                                               |
+| `textDocument/formatting`                  | ❌ Not implemented    | -                     | Document formatting                                                                                                              |
+| `textDocument/rangeFormatting`             | ❌ Not implemented    | -                     | Range-based formatting                                                                                                           |
+| `textDocument/onTypeFormatting`            | ❌ Not implemented    | -                     | Format-on-type                                                                                                                   |
```

Comment:
Lsp4j can support it, why you  say not implemented ?

Reply:
I have tested it and it did not work, however im not sure if tinymist supports it all, which is why I have left the note as unimplemented